src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -941,7 +941,7 @@ namespace winrt::TerminalApp::implementation
             // process until later, on another thread, after we've already
             // restored the CWD to it's original value.
             winrt::hstring newWorkingDirectory{ settings.StartingDirectory() };
-            if (newWorkingDirectory.size() <= 1 ||
+            if (newWorkingDirectory.size() <= 1 &&
                 !(newWorkingDirectory[0] == L'~' || newWorkingDirectory[0] == L'/'))
```

Comment:
Wouldn't this potentially crash when `newWorkingDirectory` is blank? It's going to try to dereference `newWorkingDirectory[0]` when that's technically out of bounds.

Reply:
Since c++11 it is guaranteed to just return the null terminator of the string

Reply:
That's a `winrt::hstring` though. Does that have the same guarantee?

Reply:
I mean, if it doesn't it's kind of a bug in c++/winrt isn't it?

Reply:
I don't know. You'll have to take that up with the winrt authors. It definitely looks to me like the hstring array operator expects the index to be in range.

https://github.com/microsoft/cppwinrt/blob/71832b634d12271d540383df35bde0ea1c69156c/strings/base_string.h#L243

Reply:
Oof